BuildScript.Nuke/Components/IGenerateSbom.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership.  rubicon licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.  You may obtain a copy of the
+// License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+using JetBrains.Annotations;
+using Nuke.Common;
+using Nuke.Common.ProjectModel;
+using Nuke.Common.Tooling;
+using Remotion.BuildScript.GenerateSbom;
+using Serilog;
+
+namespace Remotion.BuildScript.Components;
+
+public interface IGenerateSbom: IBuild
+{
+  [NuGetPackage(
```

Comment:
Should we pin the version of CycloneDX? would this pinning already happen via Directory.packages.props? or are we always pulling the latest version and hope for the best? Thoughts? see also comment on Cyclonedx-cli.

Reply:
Looking into Nuke's code base, this seems to check for the tool already being installed locally, so the nuget package seems to be required in a nuget package cache somewhere already. Not sure whether it also pulls the package if relevant, but it would probably just be good if we add the package manually somewhere, as on my pc I already have the tool. This would probably? fail on teamcity right now

Reply:
Ok, looking deeper: It also looks into the project.assets.json file to see which version we use, so that should work

Reply:
Okay, this means it uses the version we depend on in the csproj file of the Nuke-build-script and we need to add it there?

Reply:
I dont know. Nuke does not say whether it uses the things from the solution we want to build or from the library we built around nuke.

Reply:
Nevermind, I found it. we need to give the version of the tool in as a package reference or package download. So the things set in Buildscript.Nuke matter, the versions should be settable that way as well.